vm/systemSmartContracts/esdt.go

```diff
@@ -186,19 +200,25 @@ func (e *esdt) issue(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
 		e.eei.AddReturnMessage("not enough arguments")
 		return vmcommon.FunctionWrongSignature
 	}
-	if len(args.Arguments[0]) < minLengthForTokenName || len(args.Arguments[0]) > maxLengthForTokenName {
+	err := e.eei.UseGas(e.gasCost.MetaChainSystemSCsCost.ESDTIssue)
+	if err != nil {
+		e.eei.AddReturnMessage("not enough gas")
+		return vmcommon.OutOfGas
+	}
+	esdtConfig, err := e.getESDTConfig()
+	if err != nil {
+		e.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+	if len(args.Arguments[0]) < int(esdtConfig.MinTokenNameLength) ||
+		len(args.Arguments[0]) > int(esdtConfig.MaxTokenNameLength) {
 		e.eei.AddReturnMessage("token name length not in parameters")
 		return vmcommon.FunctionWrongSignature
 	}
-	if args.CallValue.Cmp(e.baseIssuingCost) != 0 {
+	if args.CallValue.Cmp(esdtConfig.BaseIssuingCost) != 0 {
 		e.eei.AddReturnMessage("callValue not equals with baseIssuingCost")
 		return vmcommon.OutOfFunds
```

Comment:
why not userError?

Reply:
it is similar - out of funds in general says that no enough tokens were sent